sdb/commands/zfs/vdev.py

```diff
@@ -68,6 +69,29 @@ def __init__(self,
         if self.args.weight:
             self.arg_list.append("-w")
 
+    #
+    # Iterate over the metaslabs to accumulate histogram data.
+    #
+    def sum_histograms(self,
+            metaslabs: Iterable[drgn.Object]):
+        shift = -1;
+        first_time = True
+        histsum: List[int] = [];
+        for msp in metaslabs:
+            if msp.ms_sm == sdb.get_typed_null(msp.ms_sm.type_):
+                continue
+            histogram = msp.ms_sm.sm_phys.smp_histogram;
+            if first_time:
+                shift = int(msp.ms_sm.sm_shift)
+                length = len(histogram)
+                histsum = [0] * length
+            assert length == len(histogram)
+            assert shift == int(msp.ms_sm.sm_shift)
+            for (bucket, value) in enumerate(histogram):
+                histsum[bucket] += int(value)
+            first_time = False
+        return sdb.create_object('uint64_t[64]', histsum), shift
```

Comment:
Seems like we should use `length` here instead of hardcoding `64`, or if we really depend on 64, we can hardcode/assert that `length` is 64.  It looks like `print_histogram()` uses `len(hist)` so we don't need to hardcode it here.
```suggestion
        return sdb.create_object(f'uint64_t[{length}]', histsum), shift
```

Reply:
Yup, I agree. Changed.